devito/operator.py

```diff
@@ -217,29 +185,19 @@ def arguments(self, *args, **kwargs):
                     dle_arguments[i.argument.name] = i.value(dim_size)
                 except TypeError:
                     dle_arguments[i.argument.name] = i.value
-                    # User-provided block size available, do not autotune
-                    maybe_autotune = False
             else:
                 dle_arguments[i.argument.name] = dim_size
-        dim_sizes.update(dle_arguments)
-
-        # Insert loop size arguments from dimension values
-        d_args = [d for d in arguments.values() if isinstance(d, Dimension)]
-        for d in d_args:
-            arguments[d.name] = dim_sizes[d.name]
-
-        # Might have been asked to auto-tune the block size
-        if maybe_autotune:
-            arguments = self._autotune(arguments)
-
-        # Add profiler structs
-        arguments.update(self._extra_arguments())
-
-        # Sanity check argument derivation
-        for name, arg in arguments.items():
-            if isinstance(arg, SymbolicData) or isinstance(arg, Dimension):
-                raise ValueError('Runtime argument %s not defined' % arg)
-        return arguments, dim_sizes
+        return dle_arguments
+
+    def _print_args(self, arguments, log=debug):
+        arg_str = []
+        for k, v in arguments.items():
+            if hasattr(v, 'shape'):
```

Comment:
It could be a `numpy.ndarray` or it could be from the `SymbolicData` heirarchy or it could be one of those wrapper objects you are building for yask allocated memory. The purpose of using `hasattr` instead of class names is that if it has a shape, it is a tensor so we shouldn't print the full tensor - only the L2 norm.